src/pydicom/pixels/decoders/base.py

```diff
@@ -1114,19 +1115,25 @@ def _as_array_native(runner: DecodeRunner, index: int | None) -> "np.ndarray":
             A 1D array containing the pixel data.
         """
         length_bytes = runner.frame_length(unit="bytes")
+        length_pixels = int(runner.frame_length(unit="pixels"))
         dtype = runner.pixel_dtype
 
         src: memoryview | BinaryIO
         if runner.is_dataset or runner.is_buffer:
             src = memoryview(cast(Buffer, runner.src))
             file_offset = 0
-            length_source = len(src)
+            length_source: int | float = len(src)
         else:
             src = cast(BinaryIO, runner.src)
             # Should be the start of the pixel data element's value
             file_offset = src.tell()
             length_source = length_bytes * runner.number_of_frames
 
+        # Number of bits to remove from the start and end after unpacking in
+        # the case of bits_allocated = 1
+        bit_offset_start: int | None = None
+        bit_offset_end: int | None = None
```

Comment:
I don't think an explicit `bit_offset_end` is required if `bit_offset_start` and the number of pixels is known.

Reply:
Addressed in [896c6b1](https://github.com/pydicom/pydicom/pull/2140/commits/896c6b1fc48f431d63e60d944aedd6d53a947293)